code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -1324,6 +1329,14 @@ var/global/list/damage_icon_parts = list()
 
 	if(update_icons)   update_icons()
 
+/mob/living/carbon/human/proc/update_misc_effects()
+	misc_effect_overlays.Cut()
+
+	//Begin appending miscellaneous effects.
+	if(eyes_shine())

```

Comment:
This code here is more/less placeholder while I prompt for your thoughts on how this is handled @tigercat2000. I can further abstract this by developing a light datun-based 'misc appearance effects' system so this update_misc_effects() can check a mob's misc_appearance_effects list that would have a reference to an eye_shine_effect datum thing that would have a proc called .assess() or something that would call the below checks and return the overlay.

Reply:
i personally think it would be interesting espically if we could kinda expand it further to do limb icon generation like TG does, or if it makes applying the debrained sprite i never got to work easier. by that i mean https://github.com/tgstation/tgstation/blob/bc20a758d5f2e75e93587a58c4cd0ea5009bf243/code/modules/mob/living/carbon/update_icons.dm#L254  and https://github.com/tgstation/tgstation/blob/bc20a758d5f2e75e93587a58c4cd0ea5009bf243/code/modules/surgery/bodyparts/bodyparts.dm#L208 but up to tiger (our icon code is bad and should feel bad)

Reply:
It looks a lot better now, thank you.